Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -70,6 +70,28 @@
 
 #include "UICommon/GameFile.h"
 
+namespace
+{
+class GameListTableView : public QTableView
+{
+public:
+  explicit GameListTableView(QWidget* parent = nullptr) : QTableView(parent) {}
+
+protected:
+  QModelIndex moveCursor(CursorAction cursorAction, Qt::KeyboardModifiers modifiers) override
+  {
+    // QTableView::moveCursor handles home by moving to the first column and end by moving to the
+    // last column, unless control is held in which case it ALSO moves to the first/last row.
+    // Columns are irrelevant for the game list, so treat the home/end press as if control were
+    // held.
+    if (cursorAction == CursorAction::MoveHome || cursorAction == CursorAction::MoveEnd)
+      return QTableView::moveCursor(cursorAction, modifiers | Qt::ControlModifier);
+    else
+      return QTableView::moveCursor(cursorAction, modifiers);
```

Comment:
Takes care of two issues at once for List mode; we'd need a similar GameListGridView class for Grid mode.
```suggestion
    QModelIndex index;
    if (cursorAction == CursorAction::MoveHome || cursorAction == CursorAction::MoveEnd)
      index = QTableView::moveCursor(cursorAction, modifiers | Qt::ControlModifier);
    else
      index = QTableView::moveCursor(cursorAction, modifiers);

    // Make Qt show the selected row even if it doesn't change (such as when pressing up when the
    // current index is 0), and also make navigation work when the first column is hidden.
    clearSelection();
    setCurrentIndex(index);

    return index;
```

Reply:
Note that the grid mode doesn't have the home/end issue; home/end works properly there without a workaround.  I'll see if this seems to behave better otherwise though.

Reply:
Doing this breaks multi-selection with the keyboard (both shift+arrows and control+arrows with control+space to toggle selection).  This matters because there are a few actions in the gamelist that work with multi-selection (deleting files, exporting Wii saves, and converting disc images).

Reply:
Ok, nevermind. There's probably some way to get around that but I think I've had enough of trying to work around Qt bugs for now.